src/installer/BeatPackageCompiler/AgentCustomAction.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Deployment.WindowsInstaller;
+using WixSharp;
+
+namespace Elastic.PackageCompiler.Beats
+{
+    public class AgentCustomAction
+    {
+        [CustomAction]
+        public static ActionResult MyAction(Session session)
```

Comment:
My guess is we also need a custom action for uninstall?

Reply:
yes we do. on it... not as straightforward as it seems, unfortunately.